paper/paper.md

```diff
@@ -0,0 +1,147 @@
+for use in documentation, tests, benchmarks, and workshops.
+Prominent examples in Python include scikit-learn [@scikit-learn] and
+scikit-image [@scikit-image].
+The usual approach is to include smaller datasets in the GitHub repository
+directly and package them with the source and binary distributions.
+Larger datasets require writing code to download the files from a remote server
+to the user's computer.
+The same problem is faced by scientists using version control to manage their
+research projects.
+As data files increase in size, it becomes unfeasible to store them on GitHub
+repositories.
+While downloading a data file over HTTP can be done easily with modern Python
+libraries, it is not trivial to manage a set of files, keep them updated, and
+check for corruption.
+Instead of scientists and library authors recreating the same code, it would be
+best to have a minimalistic and easy to setup tool for fetching and maintaining
+data files.
+
+Pooch is a Python library that fills this gap.
+It manages a data *registry* by downloading files from one or more remote
+servers and storing them in a local data cache.
+Pooch is written in pure Python and has minimal dependencies.
+The integrity of downloads is verified by comparing the file's SHA256 hash with
+the one stored in the data registry.
+This is also the mechanism used to detect if a file needs to be re-downloaded
+due to an update in the registry.
+Pooch is meant to be a drop-in replacement for the custom download code that
+users have already written (or are planning to write).
+In the ideal scenario, the end-user of a software package should not need to know that
+Pooch is being used.
+Setup is as easy as calling a single function (`pooch.create`), including
+setting up an environment variable for overwriting the data cache path and
+versioning the downloads so that multiple versions of the same package can
+coexist in the same machine.
+For example, this is the code required to setup a module
+`datasets.py` that uses Pooch to manage data downloads:
+
+```python
+import pooch
+
+# Get the version string from the project
+from . import version
+
+# Create a new instance of pooch.Pooch
+GOODBOY = pooch.create(
+    # Cache path using the default for the operating system
+    path=pooch.os_cache("myproject"),
+    # Base URL of the remote data server (for example, on GitHub)
+    base_url="https://github.com/me/myproject/raw/{version}/data/",
+    # PEP440 compliant version number (added to path and base_url)
+    version=version,
+    # An environment variable that overwrites the path
+    env="MYPROJECT_DATA_DIR",
+)
+# Load the registry from a simple text file.
+# Each line has: file_name sha256 [url]
+GOODBOY.load_registry("registry.txt")
+
+def fetch_some_data():
+    # Get the path to the data file in the local cache
+    # If it's not there or needs updating, download it
+    fname = GOODBOY.fetch("some-data.csv")
+    # Load it with numpy/pandas/xarray/etc
```

Comment:
```suggestion
    # Load it with NumPy/pandas/xarray/etc.
```

Reply:
Leaving this one out since all other comments don't have periods. Should I include one in all sentences?

Reply:
Sure, I only added it for the "etc" abbreviation, but don't worry about it for comments :)

Reply:
Ah, right sorry I missed that!